server/matchmaker/matchmaker_queue.py

```diff
@@ -52,14 +75,19 @@ def __init__(
             await self.timer.next_pop(lambda: len(self.queue))
 
             self.find_matches()
-            server.stats.gauge(f"matchmaker.queue.{self.queue_name}.matches", len(self._matches))
-            if self._matches:
-                server.stats.gauge(
-                    f"matchmaker.queue.{self.queue_name}.quality",
-                    mean(map(lambda m: m[0].quality_with(m[1]), self._matches))
+
+            number_of_matches = len(self._matches)
+            metrics.matches.labels(self.queue_name).set(number_of_matches)
+
+            #TODO: Move this into algorithm, then don't need to recalculate quality_with?
+            # Probably not a major bottleneck though.
+            for match in self._matches:
```

Comment:
This could be even slicker with some unpacking (parenthesis optional)
```suggestion
            for (s1, s2) in self._matches:
```